tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,250 @@
+#!groovy
+node {
+    def rootPath = "/go/src/github.com/rancher/rancher/tests/v2/validation/"
+    def job_name = "${JOB_NAME}"
+    if (job_name.contains('/')) { 
+      job_names = job_name.split('/')
+      job_name = job_names[job_names.size() - 1] 
+    }
+    def testContainer = "${job_name}${env.BUILD_NUMBER}_test"
+    def imageName = "rancher-validation-${job_name}${env.BUILD_NUMBER}"
+    def testsDir = "github.com/rancher/rancher/tests/v2/validation/${env.TEST_PACKAGE}"
+    def testResultsOut = "results.xml"
+    def envFile = ".env"
+    def rancherConfig = "rancher_env.config"
+    def branch = "release/v2.7"
+    if ("${env.BRANCH}" != "null" && "${env.BRANCH}" != "") {
+      branch = "${env.BRANCH}"
+    }
+    def rancherRepo = scm.getUserRemoteConfigs()[0].getUrl()
```

Comment:
We most likely never run into this issue as there's always going to be a value in `getUserRemoteConfigs()[0]`. But do you think a check for `null` will make this safer or is it just redundant?

Reply:
I don't have a strong opinion, but for that, the user would need to delete the default repo from the repo input, correct me if I am wrong. Also, this is unifying with Jenkinsfile.e2e. I can add this check to this, and upgrade HA and e2e Jenkinsfiles if agreed